Makefile

```diff
@@ -1,155 +1,80 @@
+# Cavium
+# Mainflux
 #
 # SPDX-License-Identifier: Apache-2.0
 #
 
-
-.PHONY: build clean test docker run
-
-
-GO=CGO_ENABLED=0 GO111MODULE=on go
-GOCGO=CGO_ENABLED=1 GO111MODULE=on go
-
-DOCKERS=docker_config_seed docker_export_client docker_export_distro docker_core_data docker_core_metadata docker_core_command docker_support_logging docker_support_notifications docker_sys_mgmt_agent docker_support_scheduler
-.PHONY: $(DOCKERS)
-
-MICROSERVICES=cmd/config-seed/config-seed cmd/export-client/export-client cmd/export-distro/export-distro cmd/core-metadata/core-metadata cmd/core-data/core-data cmd/core-command/core-command cmd/support-logging/support-logging cmd/support-notifications/support-notifications cmd/sys-mgmt-executor/sys-mgmt-executor cmd/sys-mgmt-agent/sys-mgmt-agent cmd/support-scheduler/support-scheduler
-
-.PHONY: $(MICROSERVICES)
-
-VERSION=$(shell cat ./VERSION)
-
-GOFLAGS=-ldflags "-X github.com/edgexfoundry/edgex-go.Version=$(VERSION)"
-
 GIT_SHA=$(shell git rev-parse HEAD)
+VERSION=$(shell cat ./VERSION)
 
-build: $(MICROSERVICES)
-
-cmd/config-seed/config-seed:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/config-seed
-
-cmd/core-metadata/core-metadata:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/core-metadata
-
-cmd/core-data/core-data:
-	$(GOCGO) build $(GOFLAGS) -o $@ ./cmd/core-data
-
-cmd/core-command/core-command:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/core-command
-
-cmd/export-client/export-client:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/export-client
-
-cmd/export-distro/export-distro:
-	$(GOCGO) build $(GOFLAGS) -o $@ ./cmd/export-distro
-
-cmd/support-logging/support-logging:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/support-logging
-
-cmd/support-notifications/support-notifications:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/support-notifications
+BUILD_DIR := build
 
-cmd/sys-mgmt-executor/sys-mgmt-executor:
-	$(GO) build $(GOFLAGS) -o $@ ./cmd/sys-mgmt-executor
+###
+# For ARM cross-compilation add something like:
+#
+# CC = arm-linux-gnueabi-gcc
+# CGO_LDFLAGS = -L/<path_to_libzmq>/src/.libs
+# CGO_CFLAGS = -I/<path_to_libzmq>/include
+# GOOS = linux
+# GOARCH = arm
+# GOARM = 7
+###
+FLAGS = GOOS=${GOOS} GOARCH=${GOARCH} GOARM=${GOARM} GO111MODULE=on
+GO_FLAGS =  $(FLAGS) CGO_ENABLED=0 
+CGO_FLAGS =  $(FLAGS) CC=${CC} CGO_LDFLAGS=${CGO_LDFLAGS} CGO_CFLAGS=${CGO_CFLAGS} CGO_ENABLED=1
+
+SERVICES_GO := config-seed export-client core-metadata core-command support-logging support-notifications sys-mgmt-executor sys-mgmt-agent support-scheduler
+SERVICES_CGO := core-data export-distro
+SERVICES := $(SERVICES_GO) $(SERVICES_CGO)
+DOCKERS = $(addprefix docker_,$(SERVICES))
+
+.PHONY: $(SERVICES) $(DOCKERS) build clean test docker run
+
+define compile_service
+	$(1) go build -ldflags "-s -w -X github.com/edgexfoundry/edgex-go.Version=$(VERSION)" -o cmd/$(2)/$(2) cmd/$(2)/main.go
+endef
+
+define clean_services
+	for svc in $(SERVICES); do \
+		rm -rf cmd/$$svc/$$svc; \
+	done
+endef
+
+define make_docker
+	docker build \
+		--no-cache \
```

Comment:
This line needs to be removed. It is eliminating the current caching we use between assembling Docker images which greatly improves build times in the CI/CD pipeline.